defdap/hrdic.py

```diff
@@ -175,17 +175,21 @@ def __init__(self, path, fname, dataType=None):
         # max shear component
         self.eMaxShear = np.sqrt(((self.e11 - self.e22) / 2.)**2 + self.e12**2)
 
-        # Dictionary references to all map types
-        self.component = {'f11': self.f11, 'f12': self.f12, 'f21': self.f21, 'f22': self.f22,
-                         'e11': self.e11, 'e12': self.e12, 'e22': self.e22,
-                         'eMaxShear': self.eMaxShear,
-                         'x_map': self.x_map, 'y_map': self.y_map}
-
         # crop distances (default all zeros)
         self.cropDists = np.array(((0, 0), (0, 0)), dtype=int)
 
         self.plotDefault = lambda *args, **kwargs: self.plotMaxShear(plotGBs=True, *args, **kwargs)
 
+    @property
+    def component(self):
+        # Dictionary references to all map components
+        components = {'f11': self.f11, 'f12': self.f12, 'f21': self.f21, 'f22': self.f22,
+                         'e11': self.e11, 'e12': self.e12, 'e22': self.e22,
+                         'eMaxShear': self.eMaxShear,
+                         'x_map': self.x_map, 'y_map': self.y_map}
+
+        return components
+    
```

Comment:
I don't like the idea of doing this, it means you could never add new components to the map for plotting (eventual goal for this). If there are issues with the references changing after your masking process you should try and apply the mask to the data in-place or change update the specific entries in the dict that the mask is applied to

Reply:
Fixed in 8e25626

Reply:
We could consider removing the e11, e22… variables and defining these within the component dictionary in the first place actually which would tidy things up further.

Reply:
Yeh, fine with that. We should move away from storing data directly as attributes on the objects and instead store in a dictionary with metadata included as well, one day I might make all the changes and make it more streamlined to work with data created at runtime

Reply:
I would propose a format similar to this, don't know what you think?

```
        # Dictionary to store map components
        self.component={}

        # Deformation gradient
        self.component['f11']= xDispGrad[1] + 1
        self.component['f22'] = yDispGrad[0] + 1
        self.component['f12'] = xDispGrad[0]
        self.component['f21'] = yDispGrad[1]

        # Green strain
        self.component['e11'] = xDispGrad[1] + \
                   0.5*(xDispGrad[1]*xDispGrad[1] + yDispGrad[1]*yDispGrad[1])
        self.component['e22'] = yDispGrad[0] + \
                   0.5*(xDispGrad[0]*xDispGrad[0] + yDispGrad[0]*yDispGrad[0])
        self.component['e12'] = 0.5*(xDispGrad[0] + yDispGrad[1] +
                        xDispGrad[1]*xDispGrad[0] + yDispGrad[1]*yDispGrad[0])

        # max shear component
        self.component['eMaxShear'] = np.sqrt(((self.e11 - self.e22) / 2.)**2 + self.e12**2)

```